numpy/lib/tests/test_financial.py

```diff
@@ -7,35 +9,48 @@
     )
 
 
-class TestFinancial(object):
+class TestFinancial(TestCase):
     def test_rate(self):
-        assert_almost_equal(np.rate(10, 0, -3500, 10000),
-                            0.1107, 4)
+        self.assertAlmostEqual(
+            np.rate(10, 0, -3500, 10000),
+            0.1107, 4)
+
+    def test_rate_decimal(self):
+        rate = np.rate(Decimal('10'), Decimal('0'), Decimal('-3500'), Decimal('10000'))
+        self.assertEqual(Decimal('0.1106908537142689284704527950'), rate)
 
     def test_irr(self):
         v = [-150000, 15000, 25000, 35000, 45000, 60000]
-        assert_almost_equal(np.irr(v), 0.0524, 2)
+        self.assertAlmostEqual(np.irr(v), 0.0524, 2)
         v = [-100, 0, 0, 74]
-        assert_almost_equal(np.irr(v), -0.0955, 2)
+        self.assertAlmostEqual(np.irr(v), -0.0955, 2)
         v = [-100, 39, 59, 55, 20]
-        assert_almost_equal(np.irr(v), 0.28095, 2)
+        self.assertAlmostEqual(np.irr(v), 0.28095, 2)
         v = [-100, 100, 0, -7]
-        assert_almost_equal(np.irr(v), -0.0833, 2)
+        self.assertAlmostEqual(np.irr(v), -0.0833, 2)
         v = [-100, 100, 0, 7]
-        assert_almost_equal(np.irr(v), 0.06206, 2)
+        self.assertAlmostEqual(np.irr(v), 0.06206, 2)
         v = [-5, 10.5, 1, -8, 1]
-        assert_almost_equal(np.irr(v), 0.0886, 2)
+        self.assertAlmostEqual(np.irr(v), 0.0886, 2)
 
         # Test that if there is no solution then np.irr returns nan
         # Fixes gh-6744
         v = [-1, -2, -3]
         assert_equal(np.irr(v), np.nan)
 
     def test_pv(self):
-        assert_almost_equal(np.pv(0.07, 20, 12000, 0), -127128.17, 2)
+        self.assertAlmostEqual(np.pv(0.07, 20, 12000, 0), -127128.17, 2)
```

Comment:
Please change these back - we recently stopped using `unittest.TestCase`, due to a planned move to pytest. You can use `assert_equal` in place of `assert_almost_equal`

Reply:
😄  I did not know the `assert_equal` was in there as a helper and I'm used to using the `unitest.TestCase`. I'll make the updates, thanks for pointing it out.